src/main/java/com/amazon/sqs/javamessaging/SQSSessionCallbackScheduler.java

```diff
@@ -92,7 +92,7 @@ public void run() {
         try {            
             while (true) {
                 try {
-                    if (closed) {
+                    if (closed || !session.isRunning()) {
```

Comment:
btw do we need to adapt the tests for this case?

Reply:
Given that this is not our code and we are not fully knowledgable about the code, adapting the tests will be also blind change. The reference of these changes are [this one](https://github.com/awslabs/amazon-sqs-java-messaging-lib/issues/105) and [this](https://github.com/awslabs/amazon-sqs-java-messaging-lib/issues/47) issues. But for testing this change we will run our load test and check the result there

Reply:
I rechecked and changed the PR. Tests now are passed.